pkg/services/publicdashboards/service/service.go

```diff
@@ -168,11 +168,15 @@ func (pd *PublicDashboardServiceImpl) Create(ctx context.Context, u *user.Signed
 		},
 	}
 
-	_, err = pd.store.Create(ctx, cmd)
+	affectedRows, err := pd.store.Create(ctx, cmd)
 	if err != nil {
 		return nil, ErrInternalServerError.Errorf("Create: failed to create the public dashboard: %w", err)
 	}
 
+	if affectedRows == 0 {
```

Comment:
huh. Looking at these two error checks, the message isn't materially different and we're not really telling them WHY it wasn't created if there wasn't an error.

Either we should change  the error to distinguish (no dashboard created, affected rows = 0) or something (although, I'm not really sure if that's helpful for any reason...)

or just change the check to `if err != nil || affectedRows = 0`

I'm leading toward the latter

Reply:
Only thing is, in the first case we actually have an error, in the second we don't. If we merge them into one statement, I'm still either inventing an error with no useful details, or doing an if statement to have different log statements. 

Aesthetically, I could do an `else if affectedRows == 0` to at least keep the error handling grouped up

Reply:
@jalevin check https://github.com/grafana/grafana/pull/59591/commits/26426b5c75f99c0e6ecca3e680006a0f446aa46b, something like that

Reply:
I see your point. "unknown error" could send an operator on a wild goose chase. Let's tell them what we know: failed to create: 0 rows changed, no error reported

Reply:
done https://github.com/grafana/grafana/pull/59591/commits/6054044dded73999dc9acc6056cb7c7ecc3cabc5